### DISCUSSION TITLE
Multiple (mutually exclusive) install targets for different variants of the same file?

### DISCUSSION THREAD

**Discussion by dogtopus** [SCons/scons#4331] [Q&A]
Suppose I have 2 files, `variant1/foo.txt` and `variant2/foo.txt`. I'd like to have 2 different install targets that both install them to `/usr/share/foo/foo.txt` so users could decide what variant they want to install. How would I achieve this?

Adding 2 install targets with something like

```python
install1 = env.Install('/usr/share/foo', 'variant1/foo.txt')
env.alias('install1', install1)
install2 = env.Install('/usr/share/foo', 'variant2/foo.txt')
env.alias('install2', install2)
```

don't seem to work even with 2 different envs `Copy()`ed from the same env, with the error that states "target and source lists must have same length."

**Comment by mwichmann** [comment]
I'd try using `env.Substfile` for this. SCons is always going to be unhappy in some form if you have more than one rule for a single target path.

Also be aware that a target outside the build tree (like `/usr/share`) isn't going to be selected by default - this often comes as a surprise.

**Comment by bdbaddog** [comment]
Likely you'll have to key off on one flavor or another of command line
arguments to decide which Install() is actually run.

see: https://scons.org/doc/production/HTML/scons-user.html#chap-command-line
